src/lazy/text/buffer.rs

```diff
@@ -206,6 +214,22 @@ impl<'top> TextBuffer<'top> {
         self.offset
     }
 
+    /// Returns the row position for this buffer.
+    /// _Note: Row positions are calculated based on newline characters `\n` and `\r`. `\r\n` together in this order is considered a single newline._
+    pub fn row(&self) -> usize {
+        self.row
+    }
+
+    /// Returns the column position for this buffer.
+    /// _Note: Column positions are calculated based on current offset and previous newline byte offset._
+    pub fn column(&self) -> usize {
+        self.offset - self.prev_newline_offset + 1
+    }
+
+    pub fn prev_newline_offset(&self) -> usize {
```

Comment:
I think we should probably hide the implementation details and only expose the row/col functions. Is there a specific reason this function needs to exist?

Reply:
I intended to add this for wherever buffer gets created some states are being saved in the reader implementation. so similar to how the offset is being saved we would have to do it for `row` and `prev_column_value` eventually. 
However this can be `pub(crate)`. 
Here's the saved state usage which stores offset right now but should have row and column later: https://github.com/amazon-ion/ion-rust/blob/main/src/lazy/any_encoding.rs#L479
When we will add implementation for adding this information into the `LazyValue` we will need this info and which is why I added it here. 
I am happy to remove this method entirely if thats confusing and will add it when we will require while implementing support for location with `LazyValue`.

Reply:
I think this discussion is going beyond the scope of this PR. I will just remove this method for now. And we can decide on keeping or not when it is actually required by other places.